var/spack/repos/builtin/packages/amdlibm/package.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Amdlibm(SConsPackage):
+    """AMD LibM is a software library containing a collection of basic math
+    functions optimized for x86-64 processor-based machines. It provides
+    many routines from the list of standard C99 math functions.
+    Applications can link into AMD LibM library and invoke math functions
+    instead of compiler's math functions for better accuracy and
+    performance."""
+
+    homepage = "https://developer.amd.com/amd-aocl/amd-math-library-libm/"
+    git = "https://github.com/amd/aocl-libm-ose.git"
+    maintainers = ["amd-toolchain-support"]
+
+    # Download and build from latest source
+    version("master", branch="master")
+
+    variant("debug", default=False,
+            description="Building with debug")
+    variant("verbose", default=False,
+            description="Building with verbosity")
+
+    # Mandatory dependencies
+    depends_on("python@3.6.1:", type=("build", "run"))
+    depends_on("scons@3.1.2:", type=("build"))
+    depends_on("mpfr", type=("link"))
+
+    patch('0001-libm-ose-Scripts-cleanup-pyc-files.patch')
+    patch('0002-libm-ose-prevent-log-v3.c-from-building.patch')
```

Comment:
Personally I am against patching the `master` version of a repository, in particular with long patches. Reasons for that:
- `master` is a moving target and it may not be possible to apply the patch in some time from now
- the package is from AMD... so why not pushing the changes upstream?

Reply:
AMDLibM sources will be managed within AMD, only during external customer release master branch of https://github.com/amd/aocl-libm-ose will be updated.

AMDLibM release will happen once in 6 months. For next release cycle, we will remove all the patches.

Reply:
> AMDLibM sources will be managed within AMD, only during external customer release master branch of https://github.com/amd/aocl-libm-ose will be updated.

Is there another branch where you can push these changes? If so I think using that branch instead of master as a version is a better solution that having lengthy patches applied to a moving target.

Reply:
Incorporated suggestion from tgamblin